### PR TITLE
vivaldi: widevine-support, fixed ffmpeg-support

### PR DESCRIPTION
**Pull request by betaboon**
###### Motivation for this change
I wanted widevine-support in vivaldi to enable streaming drm-content (eg. netflix)

###### Things done
- updated vivaldi-ffmpeg-codecs
- fixed how ffmpeg-codecs is installed (the way vivaldi detects it has changed)
- packages widevine-cdm from google
- extended vivaldi to allow installing with widevine support

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by NeQuissimus — APPROVED** [review]

File: pkgs/applications/networking/browsers/vivaldi/default.nix [hunk 1]
```diff
@@ -50,8 +51,8 @@ in stdenv.mkDerivation rec {
       --set-rpath "${libPath}" \
       opt/${vivaldiName}/vivaldi-bin
   '' + stdenv.lib.optionalString proprietaryCodecs ''
-    sed -i '/^VIVALDI_FFMPEG_FOUND/ a \
-    checkffmpeg "${vivaldi-ffmpeg-codecs}/lib/libffmpeg.so"' opt/${vivaldiName}/vivaldi
+    sed -i '/^if \[ "$VIVALDI_FFMPEG_FOUND/i \
+      VIVALDI_FFMPEG_FOUND=YES\nCACHED_FFMPEG=${vivaldi-ffmpeg-codecs}/lib/libffmpeg.so' opt/${vivaldiName}/vivaldi
```
Comment (eadwu) [inline pkgs/applications/networking/browsers/vivaldi/default.nix]: `opt/${vivaldiName}/vivaldi` -> `opt/${vivaldiName}/${vivaldiName}` for compatibility with snapshots. Or fixing the symlink from `opt/${vivaldiName}/${vivaldiName}` to `opt/${vivaldiName}/vivaldi`.

File: pkgs/applications/networking/browsers/vivaldi/default.nix [hunk 2]
```diff
@@ -76,7 +77,11 @@ in stdenv.mkDerivation rec {
         "$out"/share/icons/hicolor/''${d}x''${d}/apps/vivaldi.png
     done
     wrapProgram "$out/bin/vivaldi" \
-      --suffix XDG_DATA_DIRS : ${gtk3}/share/gsettings-schemas/${gtk3.name}/
+      --suffix XDG_DATA_DIRS : ${gtk3}/share/gsettings-schemas/${gtk3.name}/ \
+      ${stdenv.lib.optionalString enableWidevine "--suffix LD_LIBRARY_PATH : ${libPath}"}
+  '' + stdenv.lib.optionalString enableWidevine ''
+    rm $out/opt/vivaldi/libwidevinecdm.so
+    ln -s ${vivaldi-widevine}/lib/libwidevinecdm.so $out/opt/vivaldi/libwidevinecdm.so
```
Comment (eadwu) [inline pkgs/applications/networking/browsers/vivaldi/default.nix]: `$out/opt/vivaldi` -> `$out/opt/${vivaldiName}` for the same reason.

**Review by eadwu — COMMENTED** [review]
Nice, seems like I can stop putting everything in `.local/lib/vivaldi{,-snapshot}` now.

**Review by eadwu — APPROVED** [review]
Looks fine on my end.

**Pull request merged**